### DISCUSSION TITLE
How to get all elements of a page and delete some elemets by rectanular coordinates?

### DISCUSSION THREAD

**Discussion by nissansz** [pymupdf/PyMuPDF#2070] [Ideas]
How to get all elements of a page and delete some elemets by rectanular coordinates?

**Comment by JorjMcKie** [comment]
Use redaction annotations, `page.add_redact_annot()`.

**Reply by nissansz** [reply]
[1-2Sample.pdf](https://github.com/pymupdf/PyMuPDF/files/10053644/1-2Sample.pdf)
Check attachment for sample.
What I want to do is to
1. Replace background image of page 1
2. Remove center circled element on page 1
3.  Remove center circled element on page 2

Can you help show a sample code about how to realize the functions?

**Reply by JorjMcKie** [reply]
Not supported for this document:
This is no text, but vector graphics, which cannot be modified.

**Comment by nissansz** [comment]
Is there any way to detect vector elements? And then delete by elements?

Also, if I create pdf from exported image, how to specify page size?

p1 = fitz.open(p1)
pdfbytes = p1.convert_to_pdf()
p1pdf = fitz.open("pdf", pdfbytes)

**Reply by JorjMcKie** [reply]
> Is there any way to detect vector elements?

`page.get_drawings()`

> And then delete by elements?

As I said: no, there is not.

**Reply by nissansz** [reply]
How about redact by rect?

**Reply by JorjMcKie** [reply]
> Also, if I create pdf from exported image, how to specify page size?

Take the image width / height, then do `page = doc.new_page(width=width, height=height)`.
Then insert image `page.insert_image(page.rect, ...)`.

**Reply by nissansz** [reply]
If I want to export one page as image to stream, and then edit stream as cv2 to replace image and then send back to replace original page, any sample code?

**Reply by JorjMcKie** [reply]
No I don't have any.

**Reply by nissansz** [reply]
How to specify exported image size for each page? The size become 5950*8410 pixels when I used a code online

**Comment by nissansz** [comment]
How to print original page size?
